src/Miew.js

```diff
@@ -3593,7 +3593,13 @@ Miew.prototype._initOnSettingsChanged = function () {
   });
 
   on('fogAlpha', () => {
+    if (settings.now.fogAlpha < 0 || settings.now.fogAlpha > 1) {
```

Comment:
Miew suffers from absence of new value verification. Thank you for trying to make it better.
But I have a few comments. Direct change of settings through `settings.now `is deprecated.  We know that there is no place to learn the fact and we're sorry. And yes, `needRender` is required here. You are great!
I think the best way, for now, is to simply warn the user about invalid value. Please, add the changes and I'll accept the PR.